lib/APMSpan.js

```diff
@@ -65,7 +73,43 @@ class APMSpan {
             nodeType = 'Consumer';
         }
 
-        this[SPAN_CONTEXT].getTrace().addNode(nodeType, this, consumerCorrelationId);
+        this[SPAN_CONTEXT].getTrace().addNode(nodeType, this, consumerCorrelationId ? [{
+            value: consumerCorrelationId,
+            scope: 'Interaction',
+        }] : undefined);
+    }
+
+    _initFollowsFromSpanContext(parentReferences) {
+        function findRoot(span) {
+            let rootSpan = span;
+            while (rootSpan.getParentId()) {
+                const node = span.getTrace().findNode(rootSpan.getParentId());
+                if (node) {
+                    rootSpan = node.getSpan();
+                } else {
+                    break;
+                }
+            }
+
+            return rootSpan;
+        }
+
+        const id = utils.generateSpanId();
+        let traceId = null;
+
+        const correlationIds = [];
+        for (let i = 0; i < parentReferences.length; i += 1) {
+            const parentReference = parentReferences[i] instanceof APMSpan ? parentReferences[i].context() : parentReferences[i];
```

Comment:
Ok, for now just cause the `followsFrom` to create a correlation id to the referenced span context.

Whether a new trace is created, will be based on the lack of a childOf - so if a childOf is provided, then the new span (with the followsFrom) will just be a child of that existing trace.

I'll create a jira to update this in the java provider.